cylc/flow/scripts/cylc_psutil.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+# this file is part of the cylc suite engine.
+# copyright (c) 2008-2019 niwa
+#
+# this program is free software: you can redistribute it and/or modify
+# it under the terms of the gnu general public license as published by
+# the free software foundation, either version 3 of the license, or
+# (at your option) any later version.
+#
+# this program is distributed in the hope that it will be useful,
+# but without any warranty; without even the implied warranty of
+# merchantability or fitness for a particular purpose.  see the
+# gnu general public license for more details.
+#
+# you should have received a copy of the gnu general public license
+# along with this program.  if not, see <http://www.gnu.org/licenses/>.
+"""Report system utilisation information.
+
+For internal use with the `cylc.flow.host_select` module.
+"""
+from itertools import dropwhile
+import json
+import sys
+
+import psutil
+
+# from cylc.flow.host_select import _simple_eval
+from cylc.flow.option_parsers import OptionParser
+from cylc.flow.terminal import cli_function
+
+
+def get_option_parser():
+    return OptionParser(__doc__)
+
+
+@cli_function(get_option_parser)
+def main(parser, options):
+    # Users may have profile scripts that write to STDOUT.
+    # Drop all output lines until the the first character of a
+    # line is '['. Hopefully this is enough to find us the
+    # first line that denotes the beginning of the expected
+    # JSON data structure.
```

Comment:
I have actually seen someone with this:
```bash
# .profile
echo "[Hello me!]"
```
From testing this does result a pretty much incomprehensible error on this branch.  Could we prefix the json with a magic string, to distinguish it from spurious `[` characters?

Reply:
Wow that's bad, having said that I've seen someone with a profile something like this:

```bash
echo "Hello $USER"
echo "Today is $(date)"
echo "It's a lovely day isn't it"
echo "You have $(cylc scan | wc -l) suites running, aren't you lucky"
# yada yada
```

We could put a magic delimiter in the output, however it would mean that the output is no longer valid JSON. If we do it for this command then we should also do it for all other Cylc commands which return JSON output, scan, etc?

Another option would be to try and parse the entire stdout as JSON, if it fails remove the first line and retry.

Option three would be to give users an "environment sanity check command" which could check for this kind of nonsense, maybe check DNS etc too.

For now I think I'll have a crack at option 2?

Reply:
I guess option 2 is ok as it’s unlikely there would be many lines of garbage. But worth checking the impact if there is a lot?

Reply:
I guess option 2 is ok as it’s unlikely there would be many lines of garbage. But worth checking the impact if there is a lot?

Reply:
Done, one caveat, stdout junk must be followed by a newline (otherwise we would have to strip off one character at a time).

We could potentially write a warning to stderr when stdout junk is detected.